storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1020,11 +1049,26 @@ public static UncaughtExceptionHandler createDefaultUncaughtExceptionHandler() {
             }
         };
     }
-    
+
+    public static UncaughtExceptionHandler createWorkerUncaughtExceptionHandler() {
+        return (thread, thrown) -> {
+            try {
+                handleWorkerUncaughtException(thrown);
+            } catch (Error err) {
+                LOG.error("Received error in thread {}.. terminating worker...", thread.getName(), err);
+                Runtime.getRuntime().exit(-2);
```

Comment:
Should this just rethrow a RuntimeException so it can be handled by the parent up the hierarchy (i.e. no catch block here).

Reply:
I was just following createDefaultUncaughtExceptionHandler().  If you feel this should change, I would prefer to work as a separate JIRA.